lib/runAll.js

```diff
@@ -102,38 +101,48 @@ https://github.com/okonet/lint-staged#using-js-functions-to-customize-linter-com
     return 'No tasks to run.'
   }
 
-  return new Listr(
-    [
-      {
-        title: 'Stashing changes...',
-        skip: async () => {
-          const hasPSF = await git.hasPartiallyStagedFiles({ cwd: gitDir })
-          if (!hasPSF) {
-            return 'No partially staged files found...'
-          }
-          return false
+  const git = new GitWorkflow(gitDir)
+
+  try {
+    await new Listr(
+      [
+        {
+          title: 'Preparing...',
+          task: () => git.stashBackup()
+        },
+        {
+          title: 'Running tasks...',
+          task: () => new Listr(tasks, { ...listrOptions, concurrent: true, exitOnError: false })
         },
-        task: ctx => {
-          ctx.hasStash = true
-          return git.gitStashSave({ cwd: gitDir })
+        {
+          title: 'Applying unstaged changes...',
+          skip: ctx => ctx.hasErrors && 'Skipped because of errors from tasks',
+          task: () => git.restoreUnstagedChanges()
+        },
+        {
+          title: 'Restoring original state due to errors...',
```

Comment:
I'd drop ` due to errors` since it's quite ambiguous (what errors?)

Reply:
Errors during tasks or applying unstaged changes cause this. I'll reword.